server/client.go

```diff
@@ -1057,7 +1046,7 @@ func (c *client) flushClientsWithCheck(budget time.Duration, clientsKindOnly boo
 			continue
 		}
 
-		if budget > 0 && (!clientsKindOnly || c.kind == CLIENT && cp.kind == CLIENT) && cp.out.lft < 2*budget && cp.flushOutbound() {
+		if budget > 0 && cp.out.lft < 2*budget && cp.flushOutbound() {
```

Comment:
Should we remove budget arg all together here?

Reply:
Not sure you saw my earlier comment here: https://github.com/nats-io/nats-server/pull/2684#issuecomment-964668827.
There is a case in the server that we would want to flush in place if possible: https://github.com/nats-io/nats-server/blob/91d6531b8bdbcee0fdc1d2a4810712cdf6f76a7b/server/events.go#L411